android/src/main/java/com/pagopa/ioreactnativecrypto/X509VerificationUtil.kt

```diff
@@ -0,0 +1,428 @@
+  ): ValidationResult {
+    if (certChainBase64.isEmpty()) {
+      return ValidationResult(false, ValidationStatus.INVALID_CHAIN_PATH, "Certificate chain is empty.")
+    }
+
+    val certificateFactory: CertificateFactory
+    val trustAnchorCert: X509Certificate
+    val certificateChain: List<X509Certificate>
+    val trustAnchor: TrustAnchor
+
+    // --- 1. Decode Certificates and Trust Anchor ---
+    try {
+      certificateFactory = CertificateFactory.getInstance(CERT_TYPE)
+      val trustAnchorBytes = Base64.decode(trustAnchorCertBase64, Base64.DEFAULT)
+      trustAnchorCert = certificateFactory.generateCertificate(ByteArrayInputStream(trustAnchorBytes)) as X509Certificate
+      trustAnchor = TrustAnchor(trustAnchorCert, null)
+
+      certificateChain = certChainBase64.map { certBase64 ->
+        val certBytes = Base64.decode(certBase64, Base64.DEFAULT)
+        certificateFactory.generateCertificate(ByteArrayInputStream(certBytes)) as X509Certificate
+      }
+
+      // Check if the last certificate in the chain was issued by the trust anchor
+      val lastCertInChain = certificateChain.lastOrNull()
+        ?: return ValidationResult(false, ValidationStatus.INVALID_CHAIN_PATH, "Certificate chain is effectively empty after decoding.")
+      // Allow the chain to end with the trust anchor itself OR a cert issued by the trust anchor.
+      if (!lastCertInChain.encoded.contentEquals(trustAnchorCert.encoded) &&
+        lastCertInChain.issuerX500Principal != trustAnchorCert.subjectX500Principal) {
+        return ValidationResult(false, ValidationStatus.INVALID_TRUST_ANCHOR, "Last certificate in chain not issued by trust anchor.")
+      }
+
+    } catch (e: Exception) {
+      val (status, message) = when(e) {
+        is CertificateException -> ValidationStatus.VALIDATION_ERROR to "Failed to parse certificates: ${e.message}"
+        is IllegalArgumentException -> ValidationStatus.VALIDATION_ERROR to "Invalid Base64 encoding: ${e.message}"
+        else -> ValidationStatus.VALIDATION_ERROR to "Unexpected error preparing certificates: ${e.message}"
+      }
+      return ValidationResult(false, status, message)
+    }
+
+    // --- 2. Check Validity Periods ---
+    val now = Date()
+    for (cert in certificateChain) {
+      try {
+        cert.checkValidity(now)
+      } catch (e: CertificateExpiredException) {
+        return ValidationResult(false, ValidationStatus.CERTIFICATE_EXPIRED, "Certificate expired: ${cert.subjectX500Principal}", cert)
+      } catch (e: CertificateNotYetValidException) {
+        return ValidationResult(false, ValidationStatus.CERTIFICATE_NOT_YET_VALID, "Certificate not yet valid: ${cert.subjectX500Principal}", cert)
+      }
+    }
+
+    // --- 3. Determine if Revocation Check is Needed and Fetch CRLs ---
+    val anyCertHasCdp = certificateChain.any { hasCrlDistributionPoint(it) }
+    var crls: List<X509CRL> = emptyList()
+    val performRevocationCheck: Boolean
+
+    if (anyCertHasCdp) {
```

Comment:
you could think of adding a configuration parameter where the presence of CRL is mandatory so if not present an error is returned

Reply:
Sure, thanks for the suggestion!